Parsing/parse_wQR.py

```diff
@@ -1,77 +1,250 @@
+
+    if abs(duration_sec - vti.duration_sec) > 120.0:
+        logger.error(f"Video duration significant mismatch (real/file name):"
+                     f" {duration_sec} sec vs {vti.duration_sec} sec")
+
+    clips = []
+    qrData = {}
+
+    inRun = False
+
+    clip_start = None
+    acqNum = None
+    runNum = None
+
+    #for f in vid.iter_frames(with_times=True):
+
+    # TODO: just use tqdm for progress indication
+    iframe: int = 0
+    pos_sec: float = 0.0
+    record: QrRecord = None
+
+    while True:
+        iframe += 1
+        # pos time in ms
+        pos_sec = round((iframe-1) / fps, 3)
+        ret, frame = cap.read()
+        if not ret:
+            break
+
+        f = np.mean(frame, axis=2)  # poor man greyscale from RGB
+
+        if np.mod(iframe, 50) == 0:
+            logger.debug(f"iframe={iframe} {np.std(f)}")
+
+        #    if np.std(f) > 10:
+        #        cv2.imwrite('grayscale_image.png', f)
+        #        import pdb; pdb.set_trace()
+
+        cod = decode(f, symbols=[ZBarSymbol.QRCODE])
+        if len(cod) > 0:
+            logger.debug("Found QR code: " + str(cod));
+            assert len(cod) == 1, f"Expecting only one, got {len(cod)}"
+            data = eval(eval(str(cod[0].data)).decode('utf-8'))
+            if record is not None:
+                if data == record.data:
+                    # we are still in the same QR code record
+                    logger.debug(f"Same QR code: continue")
+                    continue
+                # It is a different QR code! we need to finalize current one
+                record = finalize_record(vti, record, iframe, pos_sec)
+            # We just got beginning of the QR code!
+            logger.debug("New QR code: " + str(data))
+            record = QrRecord()
+            record.frame_start = iframe
+            record.start_time = calc_time(vti.start_time, pos_sec)
+            record.end_time = ""
+            record.start_pos_sec = pos_sec
+            record.data = data
+        else:
+            if record:
+                record = finalize_record(vti, record, iframe, pos_sec)
+
+    if record:
+        record = finalize_record(vti, record, iframe, pos_sec)
+
+
+@click.command(help='Utility to parse video and locate integrated '
+                    'QR time codes.')
+@click.argument('path', type=click.Path(exists=True))
+@click.option('--log-level', default='INFO',
+              type=click.Choice(['DEBUG', 'INFO',
+                                 'WARNING', 'ERROR',
+                                 'CRITICAL']),
+              help='Set the logging level')
+@click.pass_context
+def main(ctx, path: str, log_level):
+    logger.setLevel(log_level)
+    logger.debug("parse_wQR.py tool")
+    logger.debug(f"Working dir      : {os.getcwd()}")
+    logger.info(f"Video full path  : {path}")
+
+    if not os.path.exists(path):
+        logger.error(f"Path does not exist: {path}")
+        return 1
+
+    do_parse(path)
```

Comment:
direct logs to stderr.

could this function yield those records as soon as figures out, and here have a loop
```suggestion
    for rec in do_parse(path):
       print(rec.to_json())
```

this way we could use `do_parse` programmatically later on if needed without going back and forth through json